### PR TITLE
Feat/affected articles

### PR DESCRIPTION
**Pull request by chaosmonster**
Add functionality to filter posts to update by affected files. The list for that is given by git diff

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -43,4 +44,6 @@ async function run() {
   }
 }
 
-run();
+
```
Comment (kotzendekrabbe) [inline scripts/index.js:47]: add comment what you are doing here (not what it does but why ;) ) 
that will help other devs to understand it better.
thaaaank you

File: scripts/lib/article.js [hunk 2]
```diff
@@ -3,8 +3,14 @@ const globby = require('globby');
 const matter = require('gray-matter');
 const { updateImagesUrls } = require('./util');
 
-async function getArticlesFromFiles(filesGlob) {
+async function getArticlesFromFiles(filesGlob, affectedFiles) {
```
Comment (kotzendekrabbe) [inline scripts/lib/article.js:6]: same as above, add context. thank you

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -45,4 +45,4 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           REPOSITORY_URL: ${{ secrets.REPOSITORY_URL }}
         run: |
-          yarn publish-to-devto
+          yarn publish-to-devto $(git diff --name-only HEAD main)
```
Comment (kotzendekrabbe) [inline .github/workflows/publish.yml:48]: awesome, thanks a bunch for the fix

File: scripts/publish.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const { commitUpdatedArticles, pushUpdatedArticles } = require('./lib/git');
 
 async function publishArticles(options) {
   try {
-    const articles = await getArticlesFromFiles(options.filesGlob);
+    const articles = await getArticlesFromFiles(options.filesGlob, options.affectedFiles);
```
Comment (kotzendekrabbe) [inline scripts/publish.js:12]: this update makes the todo in line 24 unnecessary right?
or is this something else?

**Review by kotzendekrabbe — CHANGES_REQUESTED** [review]

File: scripts/index.js (hunk 1 above)
Comment (chaosmonster) [reply scripts/index.js:47]: ✅

File: scripts/lib/article.js (hunk 2 above)
Comment (chaosmonster) [reply scripts/lib/article.js:6]: ✅

File: scripts/publish.js (hunk 4 above)
Comment (chaosmonster) [reply scripts/publish.js:12]: I think so

**Review by kotzendekrabbe — APPROVED** [review]
awesome, thanks

**Pull request merged**